### DISCUSSION TITLE
The task has actually been successfully executed in version 3.1.4, but the task status in the database has not been updated and remains in the running state.

### DISCUSSION THREAD

**Discussion by LiuHongze19920301** [apache/dolphinscheduler#14386] [Q&A]
I encountered a problem while using DolphinScheduler version 3.1.x. After the worker has successfully executed the task, the task status in the database is not updated and remains in the running state. By analyzing the source code, I sorted out the process of message status change. After the worker has completed the task, regardless of whether the execution was successful or not, the worker will send a message to the master about the task status change. This kind of message can be retried, and the worker will only remove the message from the retry message cache after the master replies with an acknowledgment. However, I found that the message cache is in memory, and I have not yet found a mechanism for message persistence. Would this kind of retry mechanism cause problems? If there were problems not limited to network-related issues causing the message previously sent by the worker to the master to fail, message retry would be required. But if the worker process restarts, the messages that need to be retried and are cached in memory will disappear, leading to a loss of task status updates. I'm seeking help from the community to analyze this problem.

**Comment by zhaohehuhu** [comment]
I encountered this issue
